### PR TITLE
:sparkles: add MixupMP & refactor mixup

### PR DESCRIPTION
**Pull request by o-laurent**
I suggest adding [MixupMP](https://proceedings.mlr.press/v238/wu24e/wu24e.pdf) to the available ensemble methods. The code is @ [apple/ml-MixupMP](https://github.com/apple/ml-MixupMP), but isn't totally complete.

I have also refactored mixup to make it more transparent in the classification routine file and reduce its length, and fixed the contributing.md file. We should probably now follow gitmoji's guildelines that are more widespread.

To Do:

- [x] Add to references
- [x] Add a parameter for the mixup method and a loss to select the ratio of mixuped samples
- [x] Add two config files (one for the ensemble and another for the single models)
- [x] Check that the WR performance match the paper's
- [ ] ~Implement mixup as a transform~
- [x] Find a way to tutorial it
- [x] improve coverage (will be done after merging)

Note concerning transforms:
Keeping Mixup as nn.Modules is now intentional: these operators are training-time, batch-level augmentations (often label-, feature-, or loss-dependent, and sometimes changing batch size), which makes them a better fit for explicit use in the training step than for dataset/transform pipelines.
=> I don't think that we should aim at making them transforms anymore

### CODE REVIEW

**Comment by o-laurent** [comment]
Experimental results when selecting for the best NLL of the run (on the test set...) on CIFAR-10 WR28-10

| Category                  | Test metric | Mixup Ensembles       | MixupMP       |
|---------------------------|-------------|-------------|-------------|
| Classification            | Acc         | 96.560%     | 97.100%     |
| Classification            | Brier       | 0.07027     | 0.04675     |
| Classification            | Entropy     | 0.58236     | 0.24441     |
| Classification            | NLL         | 0.20781     | 0.11906     |
| Calibration               | ECE         | 10.728%     | 2.955%      |
| Calibration               | aECE        | 10.725%     | 2.929%      |
| Selective Classification  | AUGRC       | 0.262%      | 0.213%      |
| Selective Classification  | AURC        | 0.317%      | 0.258%      |
| Selective Classification  | Cov@5Risk   | 100.000%    | 100.000%    |
| Selective Classification  | Risk@80Cov  | 0.412%      | 0.350%      |
| Complexity                | flops       | 6093.94 G   | 6093.94 G   |
| Complexity                | params      | 145.96 M   | 145.96 M   |

Seems pretty good to me!
